mne_bids/tests/test_write.py

```diff
@@ -2196,3 +2196,50 @@ def test_sidecar_encoding(_bids_validate):
     raw_read = read_raw_bids(bids_path)
     assert_array_equal(raw.annotations.description,
                        raw_read.annotations.description)
+
+
+@pytest.mark.parametrize('dir_name, fname, reader', test_eegieeg_data)
+@pytest.mark.filterwarnings(warning_str['channel_unit_changed'])
+def test_convert_brainvision(dir_name, fname, reader, _bids_validate):
+    """Test conversion of manufacturer data format to BrainVision.
+
+    BrainVision should correctly store data from pybv>=0.5 that
+    has different non-voltage units.
+    """
+    bids_root = _TempDir()
+    data_path = op.join(testing.data_path(), dir_name)
+    raw_fname = op.join(data_path, fname)
+
+    # the BIDS path for test datasets to get written to
+    bids_path = _bids_path.copy().update(root=bids_root, datatype='eeg')
+
+    raw = reader(raw_fname)
+    kwargs = dict(raw=raw, bids_path=bids_path, overwrite=True)
+
+    # alter some channels manually; in NK and Persyst, this will cause
+    # channel to not have units
+    raw.set_channel_types({raw.info['ch_names'][0]: 'stim'})
```

Comment:
I'm trying to use these lines to alter the "units" inside the raw file. However, in the EDF file, when I do this, things still keep them as µV, so might be something to do w/ the raw file itself? 

Otw, the rest of the tests work to validate that other units are written.